### PR TITLE
Setters

### PR DESCRIPTION
**Pull request by jeremylt**
I need to also update MAGMA a little, but this should be good.

### CODE REVIEW

File: interface/ceed.c [hunk 1]
```diff
@@ -331,7 +337,41 @@ int CeedInit(const char *resource, Ceed *ceed) {
     (*ceed)->Error = CeedErrorAbort;
   (*ceed)->refcount = 1;
   (*ceed)->data = NULL;
+
+  // Set lookup table
+  foffset foffsets[25] = {
```
Comment (jeremylt) [inline interface/ceed.c]: I will turn this `25` into a global constant.

File: interface/ceed.c [hunk 2]
```diff
@@ -365,10 +405,50 @@ int CeedSetDelegate(Ceed ceed, Ceed *delegate) {
...
+    strcat(lookupname, fname);
+  }
+
+  // Find and use offset
+  for (CeedInt i = 0; i < 25; i++) {
+    if (!strcmp(ceed->foffsets[i].fname, lookupname)) {
+      size_t offset = ceed->foffsets[i].offset;
+      size_t *fpointer;
+      fpointer = (size_t *)(object + offset);
+      *fpointer = (size_t) f;
```
Comment (jeremylt) [inline interface/ceed.c]: We don't have checking that the function has the desired inputs, but we never did at this point (setting the pointer).

File: interface/ceed-qfunction.c [hunk 3]
```diff
@@ -220,6 +228,22 @@ int CeedQFunctionGetFOCCA(CeedQFunction qf, char* *focca) {
...
+
+  @param qf              CeedQFunction
+  @param[out] f          Variable to store user function
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Advanced
+**/
+
+int CeedQFunctionGetUserFunction(CeedQFunction qf, int* (*f)) {
```
Comment (jeremylt) [inline interface/ceed-qfunction.c]: I'm not super happy with how this works, but it works.

Comment (jedbrown) [reply interface/ceed-qfunction.c]: This `f` is not a pointer to a function, just a pointer to data. You could use `void (**f)(void)` if you want a pointer to a generic function pointer.

Comment (jeremylt) [reply interface/ceed-qfunction.c]: I will give that a go. I spent quite a while fighting this particular few lines.

**Comment by jeremylt** [comment]
I think this is ready too. I am ~90% sure the MAGMA backend still works, but honestly it will need a separate PR to incorporate all of the getters/setter and streamlining after we can get #111 merged.

**Pull request merged**